src/wp-includes/post.php

```diff
@@ -6453,13 +6453,25 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 		$intermediate_dir = path_join( $uploadpath['basedir'], dirname( $file ) );
 
 		foreach ( $meta['sizes'] as $size => $sizeinfo ) {
-			$intermediate_file = str_replace( wp_basename( $file ), $sizeinfo['file'], $file );
 
-			if ( ! empty( $intermediate_file ) ) {
-				$intermediate_file = path_join( $uploadpath['basedir'], $intermediate_file );
+			// Check for alternate size mime types in the sizeinfo['sources'] array to delete.
+			if ( isset( $sizeinfo['sources'] ) && is_array( $sizeinfo['sources'] ) ) {
+				foreach ( $sizeinfo['sources'] as $mime => $properties ) {
+					$intermediate_file = str_replace( wp_basename( $file ), $properties['file'], $file );
+					if ( ! wp_delete_file_from_directory( $intermediate_file, $intermediate_dir ) ) {
+						$deleted = false;
+					}
```

Comment:
Aren't we missing the following `path_join` call here? Don't the relative paths in `$sizeinfo['sources']` match the format used in `$sizeinfo['file']`?
```suggestion
					$intermediate_file = str_replace( wp_basename( $file ), $properties['file'], $file );
					if ( ! empty( $intermediate_file ) ) {
						$intermediate_file = path_join( $uploadpath['basedir'], $intermediate_file );
						if ( ! wp_delete_file_from_directory( $intermediate_file, $intermediate_dir ) ) {
							$deleted = false;
						}
					}
```

Reply:
Agree these should all be constructed the same way as the data should match, I'll apply your suggestion and double check the functionality.